### DISCUSSION TITLE
Cesium ION point cloud LAZ

### DISCUSSION THREAD

**Discussion by bortomar-intellmaps** [visgl/loaders.gl#2965] [Q&A] [answered]
Hi, we use loaders.gl 3.4.6 and recently we are facing error `A 3D tile failed to load: undefined Cannot read properties of undefined (reading 'BYTES_PER_ELEMENT')` when loading point cloud downloaded from Cesium ION asset.

![image](https://github.com/visgl/loaders.gl/assets/135999694/15fe0050-119d-482e-88e6-1b076990a33d)

That asset is at first imported to Cesium as archive  as an archive composed of laz and traj files:
![image](https://github.com/visgl/loaders.gl/assets/135999694/5e9f09f4-a845-478f-bdc5-66ce6efc9c78)
![image](https://github.com/visgl/loaders.gl/assets/135999694/99f86dfb-317a-4835-98ed-373d7c35f080)

Then we download the asset and upload it to locacl S3 minio storage from which it is served to deck.gl instance. 
![image](https://github.com/visgl/loaders.gl/assets/135999694/83ff0fdf-8898-4b6a-b469-511f5ba0ebbe)

As I wrote, the problem appeared relatively recently and concerns archives with the mentioned content (laz, traj files), which previously had no problem. We have not changed the way we create the source data or the technologies used.

Is it some change in the data import on Cesium that we have not noticed and that causes this error in the current version of loaders.gl?

Thanks for all your suggestions.

**Comment by ibgreen** [comment, marked as answer]
The discussion in https://github.com/visgl/loaders.gl/pull/2949 comes to mind, as Cesium recently started to export 64 bit attributes from LAS files into PNTS tiles and also made some non-obvious choices about how to represent this in Draco encoding. 

Assuming the issue is a same, there is a workaround fix for this in loaders.gl 4.2. 

It should not be too hard to back port it to loaders.gl 3 should you wish to put up a PR: https://github.com/visgl/loaders.gl/pull/2951

**Reply by ibgreen** [reply]
The PR would have to be opened against `3.4-release` branch

**Reply by bortomar-intellmaps** [reply]
Hi, thanks for you suggestion but still not working. I try to clone loaders.gl switch to release-3.4, make changes #2951 , build and copy `modules/draco/dist` intto my project's `node_modules/@loaders.gl/draco/dist` but nothing changed. I got same error.

![image](https://github.com/visgl/loaders.gl/assets/135999694/f375cb54-2033-4f37-a1be-78e310246ea0)

**Reply by ibgreen** [reply]
Hmm it may be loading a pre-built worker from unpkg.com. You can check your network tab in your browser dev tools to see if that is happening. You could set the workerUrl in your DracoLoader options to point to your local worker.

**Reply by ibgreen** [reply]
Tried to put up a patch for 3.4, Unfortunately 3.4 build environment seems to have experienced some rot that could take more time to fix: #3006.

**Reply by bortomar-intellmaps** [reply]
Many thanks! It is working with local worker. I made PR: https://github.com/visgl/loaders.gl/pull/3007

**Reply by bortomar-intellmaps** [reply]
Sorry, I didn't realize you already made PR. I close #3007 .
